### PR TITLE
Compilation message with MPI extension

### PR DESCRIPTION
**Pull request by ranocha**
I wanted to draft a version with a package extension for MPI so that we can still include the info message 

```
[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.
```

However, the first naive approach doesn't work nicely since we're not allowed to overwrite method definitions while precompiling. Thus, I needed to deactivate precompilation for the extension, resulting in the ugly messages

```julia
julia> using MPI, TrixiBase
[ Info: Precompiling TrixiBaseMPIExt [722ac391-ce67-58c0-be25-224aa07e4d5d]
[ Info: Skipping precompilation since __precompile__(false). Importing TrixiBaseMPIExt [722ac391-ce67-58c0-be25-224aa07e4d5d].

julia> trixi_include("test/dummy.jl")
[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.
Hello
```

We can not use something in `__init__` since that will be called before Trixi.jl is initialized - i.e., before MPI is initialized in general.

### CODE REVIEW

File: test/dummy.jl
Comment (ranocha) [inline test/dummy.jl:1]: This is a dummy file that could be used for testing `trixi_include`

File: src/trixi_include.jl
Comment (sloede) [inline src/trixi_include.jl:43]:
```suggestion
    if !mpi_isparallel(Val(:MPIExt))
```

File: ext/TrixiBaseMPIExt.jl
Comment (sloede) [inline ext/TrixiBaseMPIExt.jl:13]:
```suggestion
function TrixiBase.mpi_isparallel(::Val{:MPIExt})
```

**Review by sloede — COMMENTED** [review]
Could this work (untried) and is not too dirrrty?

File: ext/TrixiBaseMPIExt.jl
Comment (sloede) [inline ext/TrixiBaseMPIExt.jl:2]: I distinctly remember a conversation with one of the Pkg.jl gurus (Frederik?) where it was pointed out that the name of the package extension is never shown without the package itself, thus it is best practice to just name it `PackageExt`:
```suggestion
module MPIExt
```

**Review by sloede — CHANGES_REQUESTED** [review]

File: ext/TrixiBaseMPIExt.jl
Comment (ranocha) [reply ext/TrixiBaseMPIExt.jl:2]: As you see in my first post above, you can get messages like

```
[ Info: Precompiling TrixiBaseMPIExt [722ac391-ce67-58c0-be25-224aa07e4d5d]
```

Thus, I would like to keep it as it is.

Comment (ranocha) [reply ext/TrixiBaseMPIExt.jl:2]: This naming convention is also chosen in the official docs at https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)

Comment (sloede) [reply ext/TrixiBaseMPIExt.jl:2]: > This naming convention is also chosen in the official docs

I know. I _did_ have a conversation with them about it, but it seems you're right. Thus, I'm ok with leaving it as it is 👍

**Review by sloede — APPROVED** [review]
LGTM, thanks! If @efaulhaber approves, this can be merged imho (don't mind the missing format check, that's dependent on another PR be to merged)

**Review by sloede — APPROVED** [review]

File: test/test_util.jl
Comment (sloede) [inline test/test_util.jl:72]: I just noticed - this is a different style than in Trixi.jl (`force = true`). Is this a deliberate deviation from SciML, @efaulhaber?

Comment (efaulhaber) [reply test/test_util.jl:72]: I copied this from TrixiParticles, where we did not change the style to the one with the whitespaces when SciML did and instead stayed with the more common Julia style.

Comment (ranocha) [reply test/test_util.jl:72]: https://github.com/trixi-framework/TrixiBase.jl/pull/7

Comment (sloede) [reply test/test_util.jl:72]: OK, let's move the discussion to #7 then.

Comment (ranocha) [reply test/test_util.jl:72]: I'm not sure whether it's more common to have spaces around kwargs or not 🤷

File: test/test_util.jl
Comment (efaulhaber) [inline test/test_util.jl:43]: Can we somehow export this as well when importing TrixiBase from a testing environment? Then we can avoid duplicating this function from Trixi.

Comment (sloede) [reply test/test_util.jl:43]: @ranocha?

Comment (sloede) [reply test/test_util.jl:43]: @efaulhaber Maybe we can move this convo to a separate issue such that we can merge this PR (once you approve) and then start the registration process?

**Comment by sloede** [comment]
@ranocha I think the error message is not yet properly filtered out:
https://github.com/trixi-framework/TrixiBase.jl/actions/runs/7711136543/job/21015911732?pr=5#step:7:65

File: test/test_util.jl
Comment (ranocha) [reply test/test_util.jl:43]: We would need to move this macro to the plain source code of the package and make TrixiBase.jl depend on Test.

Comment (efaulhaber) [reply test/test_util.jl:43]: So there is no nice way to do it? Maybe add Test as a soft dependency or so?

Comment (ranocha) [reply test/test_util.jl:43]: As @sleode said, I'm fine discussing this further in an issue and merging this soon to start the registration process.

Comment (ranocha) [reply test/test_util.jl]: I tend to prefer moving this to a TrixiTest.jl package to avoid depending on Test etc. for everything

**Comment by ranocha** [comment]
> @ranocha I think the error message is not yet properly filtered out: https://github.com/trixi-framework/TrixiBase.jl/actions/runs/7711136543/job/21015911732?pr=5#step:7:65

Should be fixed

**Review by sloede — APPROVED** [review]

File: test/test_util.jl
Comment (sloede) [reply test/test_util.jl]: Moved to #9. Your points are absolutely valid @efaulhaber, I just wanted to make sure we can move forward with the current state.

**Pull request merged**
